app/webpack/components/authentication/log_in.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <div id="authentication-log-in">
+    <div class="loader" v-if="loading">loading...</div>
+    <div ref="firebaseAuthContainer"></div>
+  </div>
+</template>
+
+<script>
+import firebase from 'firebase/app';
+import 'firebase/auth';
+import firebaseui from 'firebaseui';
+import 'firebaseui/dist/firebaseui.css';
+import 'axios';
+import Axios from 'axios';
+export default {
+  created() {
+    this.ui = new firebaseui.auth.AuthUI(firebase.auth());
+    this.csrf_token = document
+      .querySelector('meta[name="csrf-token"]')
+      .getAttribute('content');
+  },
+  mounted() {
+    let $self = this;
+    this.uiConfig = {
+      signInOptions: [
+        {
+          provider: firebase.auth.EmailAuthProvider.PROVIDER_ID,
+          requireDisplayName: true
+        }
+      ],
+      credentialHelper: firebaseui.auth.CredentialHelper.NONE,
+      callbacks: {
+        signInSuccessWithAuthResult(authResult, redirectUrl) {
+          firebase
+            .auth()
+            .currentUser.getIdToken(true)
+            .then(idtoken => {
+              Axios.post(
```

Comment:
Need to `return …` the promise from `Axios.post` (to better handle the promise chain).

Reply:
Why? We are already handling the post response in the `then...catch`